bazaarvoice/emodb
# Changed Emo health check to run per cluster and reduced query footprint.

## Github Issue #

None

## What Are We Doing Here?

To reduce the weight of the Cassandra health check this PR changes them such that:

- Only one health check is run per cluster, not per keyspace
- The health check by default queries system.peers instead of one of the placement's tables.  This is beneficial if querying that table is being non-responsive, such as when Cassandra is under unusually heavy load.

## How to Test and Verify

1. Check out this PR
2. Call `locahost:8081/healthcheck`
3. Verify health check is present and accurate

## Risk

### Level 

`Low`

### Required Testing

`Regression`

### Risk Summary

By not querying on each keyspace's tables there is a possibility that the ring is healthy but there is a problem with the table, such as it having been dropped.  However, this is extremely unlikely since it requires proactive maliciousness to occur.

## Code Review Checklist


build and run.

we inevitably come back to this code it will only take hours to figure out, not
days.

a victim of rampaging consistency, and should be using this course of action. 
We don't have coding standards out yet for this project, so please make sure to address any feedback regarding STYLE so the codebase remains consistent.
